tools/cloud-build/babysit_tests.py

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python3
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Sequence, Dict, Tuple, Callable
+from dataclasses import dataclass
+from google.cloud.devtools import cloudbuild_v1
+from google.cloud.devtools.cloudbuild_v1.types.cloudbuild import Build, ApproveBuildRequest, ApprovalResult,RetryBuildRequest
+import time
+import argparse
+import subprocess
+
+Selector = Callable[[Build], bool]
+
+
+def selector_by_name(names) -> Selector:
+    def selector(build):
+        return any(trig_name(build) == n for n in names)
+    return selector
+
+
+SELECTORS: Dict[str, Selector] = {
```

Comment:
nit: My preference would be to move `SELECTORS` and `Status` to the top of the file to live near `Selector`.